x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -835,4 +867,241 @@ export class ManifestManager {
       );
     }
   }
+
+  /**
+   * Unified Manifest methods
+   */
+
+  private setNewSemanticVersion(semanticVersion: string): string | null {
+    const newSemanticVersion = semver.inc(semanticVersion, 'patch');
+    if (!semver.valid(newSemanticVersion)) {
+      throw new Error(`Invalid semver: ${newSemanticVersion}`);
+    }
+    return newSemanticVersion;
+  }
+
+  protected getUnifiedManifestClient(): UnifiedManifestClient {
+    return new UnifiedManifestClient(this.savedObjectsClient);
+  }
+
+  public async getAllUnifiedManifestsSOFromCache(): Promise<InternalUnifiedManifestSchema[]> {
+    // Fetch Unified Manifests from SO on getLatestManifest, cache them and remove on successful commit.
+    // This is to avoid fetching Unified Manifests from SO on every getLatestManifest call which is every task run (1 minute).
```

Comment:
`getLastestManifest()` gets called on each task execution which is every one minute. With my changes we are looking at `find` method on SO that will pull all manifest SOs (one per policy). I believe it is worth caching them, what do you think? 
I have added 'clearCache' calls on caught errors in all place that I believe could be affected - 26f35e80a2bab0104c3ed8bea943ae84ab13a4ca 
Thanks for pointing this out!

Reply:
I've decided to remove this mechanism for the time being. I've introduced it due to the fact that when updating Artifacts we would be fetching a complete list two times - once on task start when we compare them with existing ones and second time in the same task when commiting changes. However, since this won't be occurring every task run after some consideration I came to the conclusion, that maintaining this cache could potentially introduce edge cases that would be hard to think ahead of.